Gemfile

```diff
@@ -0,0 +1,8 @@
+source 'https://rubygems.org'
+ruby '2.5.0'
+
+gem 'rails', '~> 5.2.0'
+
+group :development do
```

Comment:
When developing gems, I think you should add dependencies in the gemspec instead of the Gemfile

Reply:
I might agree (Do not have much idea about it), but I believe this gem particularly is for testing our own code on this gem, and not as a dependency for future users of the gem. Is it the same anyway?

Reply:
for rails you can use `add_dependency` and for rubocop `add_development_dependency`

https://apidock.com/ruby/Gem/Specification/add_dependency
https://apidock.com/ruby/Gem/Specification/add_development_dependency

Reply:
We have found examples and read that the dependencies of a gem should be placed in the gemfile.

Reply:
I digged a bit in the [rails gem itself](https://github.com/rails/rails). It uses both a [Gemfile](https://github.com/rails/rails/blob/master/Gemfile) and has [dependencies defined on the gemspec](https://github.com/rails/rails/blob/master/rails.gemspec). What I am failing to realize is what differs between these two, why it uses both instead of one or the other.

Reply:
I think this is explained in the post I shared above:

```
If you find that you need to develop against a gem that hasn't yet been released (for instance, Rails often develops against unreleased Rack, Arel, or Mail gems), you can add individual lines in the Gemfile that tell bundler where to find the gems. It will still use the dependencies listed in the .gemspec for dependency resolution, but it now knows exactly where to find the dependency during gem development.
```

Reply:
That's not exactly what I see on the Rails Gemfile, I can see gems being declared normally there. 

In any case, I added the dependencies on the `.gemspec` file as you suggested and added a reference to the `.gemspec` file on the Gemfile .

Reply:
Looking at the Gemfile.lock of a project I see
```
    rails (5.1.6)
      actioncable (= 5.1.6)
      actionmailer (= 5.1.6)
      actionpack (= 5.1.6)
      actionview (= 5.1.6)
      activejob (= 5.1.6)
      activemodel (= 5.1.6)
      activerecord (= 5.1.6)
      activesupport (= 5.1.6)
      bundler (>= 1.3.0)
      railties (= 5.1.6)
      sprockets-rails (>= 2.0.0)
```
which match with the ones in gemspec, not in the Gemfile. I think they could be using the Gemfile for other purposes. For example in `devise` it seems they use the `Gemfile` for testing purposes. https://github.com/plataformatec/devise/blob/master/.travis.yml

Reply:
Found this article about gemfiles and gemspec https://medium.com/@divya.n/gemfile-vs-gemspec-ee72512da246

Reply:
Thanks Nico. To be honest I am more confident in the post I shared that is from Yehuda Katz, Rails core member.